frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -105,4 +123,14 @@ class SlideInDialog
       startTrackingValidation()
     }
   }
+
+  private fun isDescendant(
+      presumedDescendantName: String, presumedAncestorEntry: IBranchLayoutEntry
+  ): Boolean {
+    if (presumedAncestorEntry.children.exists { it.name.equals(presumedDescendantName) }) {
+      return true
+    } else {
+      return presumedAncestorEntry.children.exists() { isDescendant(presumedDescendantName, it) }
```

Comment:
...then it'd be just `presumedAncestorEntry.children.exists() { isDescendantOf(presumedDescendantName) }`, I guess